examples/tree_2d_dgsem/elixir_shallowwater_twolayer_well_balanced.jl

```diff
@@ -0,0 +1,78 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Semidiscretization of the two-layer shallow water equations with a bottom topography function 
+# to test well-balancedness
+
+equations = ShallowWaterTwoLayerEquations2D(gravity_constant=9.81, H0=0.6, rho_upper=0.9, rho_lower=1.0)
+
+# An initial condition with constant total water height, zero velocities and a bottom topography to 
+# test well-balancedness
+function initial_condition_well_balanced(x, t, equations::ShallowWaterTwoLayerEquations2D)
+  H_lower = 0.5
+  H_upper = 0.6
+  v1_upper = 0.0
+  v2_upper = 0.0
+  v1_lower = 0.0
+  v2_lower = 0.0
+  b = (((x[1] - 0.5)^2 + (x[2] - 0.5)^2) < 0.04 ? 0.2 * (cos(4 * pi * sqrt((x[1] - 0.5)^2 + (x[2] +
+        -0.5)^2)) + 1) : 0)
```

Comment:
Type instability
```suggestion
  b = (((x[1] - 0.5)^2 + (x[2] - 0.5)^2) < 0.04 ? 0.2 * (cos(4 * pi * sqrt((x[1] - 0.5)^2 + (x[2] +
        -0.5)^2)) + 1) : 0.0)
```

Reply:
Fixed in commit 4fd850aa42033387973d2b5a9338294706e416cc